internal/provider/packer/resources/bucket/resource_packer_bucket.go

```diff
@@ -0,0 +1,244 @@
+	modifiers.ModifyPlanForDefaultProjectChange(ctx, r.client.Config.ProjectID, req.State, req.Config, req.Plan, resp)
+}
+
+type bucket struct {
+	ProjectID      types.String `tfsdk:"project_id"`
+	OrganizationID types.String `tfsdk:"organization_id"`
+	Name           types.String `tfsdk:"name"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+}
+
+func (r *resourcePackerBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan bucket
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+	name := plan.Name.ValueString()
+	res, err := packerv2.CreateBucket(ctx, r.client, loc, name)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating bucket", err.Error())
+		return
+	}
+
+	plan.ResourceName = types.StringValue(res.ResourceName)
+	plan.ProjectID = types.StringValue(res.Location.ProjectID)
+	plan.OrganizationID = types.StringValue(res.Location.OrganizationID)
+	plan.CreatedAt = types.StringValue(res.CreatedAt.String())
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourcePackerBucket) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state bucket
+
+	// Read Terraform prior state into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resourceName := state.ResourceName.ValueString()
+	resourceNameRegex := `packer\/project\/.*\/bucket\/.*`
+	// packer/project/{project_id}/bucket/{bucket_name}
+
+	resourceParts := strings.SplitN(resourceName, "/", 5)
+	resourceNameValid, err := regexp.MatchString(resourceNameRegex, resourceName)
+	if err != nil {
+		resp.Diagnostics.AddError("Unexpected provider error", "Failed to parse regular expression, this is an internal error, please report this issue to the provider developers")
+		return
+	}
+	if !resourceNameValid || len(resourceParts) != 5 {
+		resp.Diagnostics.AddError("Invalid resource name", "Resource name expected to match packer/project/{project_id}/bucket/{bucket_name}")
+		return
+	}
+	params := packerservice.NewPackerServiceGetBucketParams()
+	params.SetLocationOrganizationID(state.OrganizationID.ValueString())
+
+	bucketName := resourceParts[4]
+
+	params.SetBucketName(bucketName)
+	bucketResp, err := r.client.PackerV2.PackerServiceGetBucket(params, nil)
+
+	projectID := resourceParts[2]
+	// TODO Does this make sense, should we use the projectID from the resource name?
```

Comment:
I don't see a reason for not using that information from the state directly. The logic for parsing the resource name seems unnecessary to get both the project ID and bucket name.

Reply:
Same as above but this is because this function is called on import too, so it has to read from the resource name if we want to import via resource name